pkg/telegram-bot/telegram-bot.go

```diff
@@ -1 +1,421 @@
 package telegrambot
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"time"
+
+	"github.com/joho/godotenv"
+	"gopkg.in/telebot.v4"
+)
+
+const (
+	USER       = "User"
+	ADMIN      = "Admin"
+	SUPERADMIN = "SuperAdmin"
+)
+
+type user struct {
+	userName string
+	password string
+	userType string
+	email    string
+}
+
+var (
+	bot *telebot.Bot
+
+	menu     = &telebot.ReplyMarkup{ResizeKeyboard: true}
+	selector = &telebot.ReplyMarkup{RemoveKeyboard: true}
+
+	data    = make(map[int64]user)
+	message = make(map[int64]*telebot.Message)
+
+	// inline btn
+	register = selector.Data("Register", "Register")
+	login    = selector.Data("Login", "Login")
+
+	profile    = selector.Data("Profile", "Profile")
+	bookmark   = selector.Data("Bookmark", "Bookmark")
+	monitoring = selector.Data("Monitoring", "Monitoring")
+	post       = selector.Data("Post", "Post")
+	export     = selector.Data("Export", "Export")
+	report     = selector.Data("Report", "Report")
+	logout     = selector.Data("Logout", "Logout")
+
+	monitoringUser      = selector.Data("Monitoring", "MonitoringUser")
+	usr                 = selector.Data("User", "User")
+	accessToUserAccount = selector.Data("Access To User Account", "AccessToUserAccount")
+
+	setting = selector.Data("Setting", "Setitng")
+	admin   = selector.Data("Admin", "Admin")
+
+	globalBookmark = selector.Data("Global Bookmarks", "Global_Bookmarks")
+	myBookmark     = selector.Data("My Bookmarks", "My_Bookmarks")
+	newPost        = selector.Data("Create New Post", "Create_new_post")
+	back           = selector.Data("Back", "Back")
+	refresh        = selector.Data("Refresh", "Refresh")
+	back_repost    = selector.Data("Back", "Back_repost")
+	skip           = selector.Data("Skip", "Skip")
+
+	// reply btn
+	cancle = menu.Text("Cancle")
+
+	// state
+	loginStatus    = false
+	registerStatus = false
+)
+
+func initServer() (*telebot.Bot, error) {
+	err := godotenv.Load("/tmp/.env")
+	if err != nil {
+		return nil, err
+	}
+
+	pref := telebot.Settings{
+		Token:  os.Getenv("TELEGRAM_BOT_TOKEN"),
+		Poller: &telebot.LongPoller{Timeout: 10 * time.Second},
+	}
+
+	bot, err = telebot.NewBot(pref)
+	if err != nil {
+		log.Fatal(err)
+		return nil, err
+	}
+
+	return bot, nil
+}
+
+func StartBot() {
```

Comment:
please write a struct of bot and make this functions as receiver function for the struct